packages/turf-kinks/index.ts

```diff
@@ -40,114 +41,16 @@ export default function kinks<T extends LineString | MultiLineString | Polygon |
-                            i === 0 &&
-                            k === line1.length - 2 &&
-                            // lineString is closed
-                            line1[i][0] === line1[line1.length - 1][0] &&
-                            line1[i][1] === line1[line1.length - 1][1]
-                        ) {
-                            continue;
-                        }
-                    }
-
-                    const intersection: any = lineIntersects(line1[i][0], line1[i][1], line1[i + 1][0], line1[i + 1][1],
-                        line2[k][0], line2[k][1], line2[k + 1][0], line2[k + 1][1]);
-                    if (intersection) {
-                        results.features.push(point([intersection[0], intersection[1]]));
-                    }
-                }
-            }
-        });
-    });
-    return results;
-}
-
-// modified from http://jsfiddle.net/justin_c_rounds/Gd2S2/light/
-function lineIntersects(
-    line1StartX: any,
-    line1StartY: any,
-    line1EndX: any,
-    line1EndY: any,
-    line2StartX: any,
-    line2StartY: any,
-    line2EndX: any,
-    line2EndY: any) {
-    // if the lines intersect, the result contains the x and y of the
-    // intersection (treating the lines as infinite) and booleans for whether
-    // line segment 1 or line segment 2 contain the point
-    let denominator;
-    let a;
-    let b;
-    let numerator1;
-    let numerator2;
-    const result = {
-            x: null,
-            y: null,
-            onLine1: false,
-            onLine2: false,
-        };
-    denominator = ((line2EndY - line2StartY) * (line1EndX - line1StartX)) - ((line2EndX - line2StartX) * (line1EndY - line1StartY));
-    if (denominator === 0) {
-        if (result.x !== null && result.y !== null) {
-            return result;
-        } else {
-            return false;
-        }
-    }
-    a = line1StartY - line2StartY;
-    b = line1StartX - line2StartX;
-    numerator1 = ((line2EndX - line2StartX) * a) - ((line2EndY - line2StartY) * b);
-    numerator2 = ((line1EndX - line1StartX) * a) - ((line1EndY - line1StartY) * b);
-    a = numerator1 / denominator;
-    b = numerator2 / denominator;
-
-    // if we cast these lines infinitely in both directions, they intersect here:
-    result.x = line1StartX + (a * (line1EndX - line1StartX));
-    result.y = line1StartY + (a * (line1EndY - line1StartY));
-
-    // if line1 is a segment and line2 is infinite, they intersect if:
-    if (a >= 0 && a <= 1) {
-        result.onLine1 = true;
-    }
-    // if line2 is a segment and line1 is infinite, they intersect if:
-    if (b >= 0 && b <= 1) {
-        result.onLine2 = true;
-    }
-    // if line1 and line2 are segments, they intersect if both of the above are true
-    if (result.onLine1 && result.onLine2) {
-        return [result.x, result.y];
-    } else {
-        return false;
+    const intersections = findIntersections(featureIn)
```

Comment:
This seems a lot cleaner!

Reply:
There is similiar logic in the sweepline-intersections library - but it does make our package look neater :)